### PR TITLE
Clean up SyncPlrKill

### PR DESCRIPTION
**Pull request by AJenbo**
Bin exact

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -2116,40 +2116,25 @@ void __fastcall DropHalfPlayersGold(int pnum)
 
 void __fastcall SyncPlrKill(int pnum, int earflag)
 {
-	int v2; // esi
-	int v3; // ebx
-	int v4; // edx
-	int v5; // eax
+	if ( !plr[pnum]._pHitPoints && !currlevel ) {
```
Comment (mewmew) [inline Source/player.cpp]: I would prefer `plr[pnum]._pHitPoints == 0 && currlevel == 0` if that is also binary exact.

File: Source/player.cpp [hunk 2]
```diff
@@ -2116,40 +2116,25 @@ void __fastcall DropHalfPlayersGold(int pnum)
...
-		else
-		{
-			while ( 1 )
-			{
-				v5 = missileactive[v4];
-				if ( missile[v5]._mitype == 13 && missile[v5]._misource == pnum && !missile[v5]._miDelFlag )
-					break;
-				if ( ++v4 >= nummissiles )
-					goto LABEL_9;
+	int ma;
```
Comment (mewmew) [inline Source/player.cpp]: Is it possible to move `int ma` to `ma = missileactive[i];` inside the loop? Only if binary exact of course.

File: Source/player.cpp [hunk 3]
```diff
@@ -2116,40 +2116,25 @@ void __fastcall DropHalfPlayersGold(int pnum)
...
+	for ( int i = 0; i < nummissiles; i++ ) {
+		ma = missileactive[i];
+		if ( missile[ma]._mitype == MIS_MANASHIELD && missile[ma]._misource == pnum && !missile[ma]._miDelFlag ) {
```
Comment (mewmew) [inline Source/player.cpp]: I'd prefer `missile[ma]._miDelFlag == 0`

File: Source/player.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/player.cpp]: Oh, I though that would give a redefined error :)

**Pull request merged**